components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -173,21 +183,30 @@ private void valiateTransactionContext(JWTClaimsSet claimsSet) throws CibaAuthFa
     private void validateBindingMessage(JWTClaimsSet claimsSet) throws CibaAuthFailureException {
 
         try {
-            // Validation for binding_message.
-            if ((claimsSet.getClaim(CibaConstants.BINDING_MESSAGE)) == null) {
-                // Request has claim for binding_message.
-                return;
-            }
 
-            if (StringUtils.isBlank(claimsSet.getStringClaim(CibaConstants.BINDING_MESSAGE))) {
-                // Binding_message with a blank value which is not acceptable.
-                if (log.isDebugEnabled()) {
-                    log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
-                            claimsSet.getIssuer() +
-                            ".The request is with invalid  value for (binding_message).");
+            // Request has claim for binding_message(optional).
+            if (claimsSet.getClaim(CibaConstants.BINDING_MESSAGE) != null) {
+                // Validate binding message is not blank
+                String bindingMessage = claimsSet.getStringClaim(CibaConstants.BINDING_MESSAGE);
+                if (StringUtils.isBlank(bindingMessage)) {
+                    // Binding_message with a blank value which is not acceptable.
+                    if (log.isDebugEnabled()) {
+                        log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                                claimsSet.getIssuer() +
+                                ".The request is with invalid  value for (binding_message).");
+                    }
+                    throw new CibaAuthFailureException(ErrorCodes.INVALID_BINDING_MESSAGE,
+                            "Invalid value for (binding_message).");
+                } else {
```

Comment:
We can remove the else branch to improve readability since If block throws an exception
WDYT?

Reply:
+1. Removed